### ISSUE TITLE
Bug: Build Error on NRF Board, Undeclared Variable 'MP_QSTR___del___'

### ISSUE THREAD

**Issue by nawalter** [micropython/micropython#11298]
I encountered an error while building Micropython on the Nordic pca10056 nrf board. Below is the error message:

```
..\micropython\ports\nrf>make BOARD=pca10056
Use make V=1 or set BUILD_VERBOSE in your environment to increase build verbosity.
GEN build-pca10056/genhdr/mpversion.h
CC ../../py/gc.c
../../py/gc.c: In function 'gc_sweep':
../../py/gc.c:393:72: error: 'MP_QSTR___del__' undeclared (first use in this function); did you mean 'MP_QSTR___new__'?
  393 |                             mp_load_method_maybe(MP_OBJ_FROM_PTR(obj), MP_QSTR___del__, dest);
      |                                                                        ^~~~~~~~~~~~~~~
      |                                                                        MP_QSTR___new__
../../py/gc.c:393:72: note: each undeclared identifier is reported only once for each function it appears in
See https://github.com/micropython/micropython/wiki/Build-Troubleshooting
make: *** [../../py/mkrules.mk:83: build-pca10056/py/gc.o] Error 1
```

I'm using the following environment:
- micropython commit: b525f1c
- OS: Windows 10
- Python: Python 3.10.11
- Toolchain: GNU Arm Embedded Toolchain 10-2020-q4-major
- Board: pca10056 (nRF52840-DK)

Please let me know if you need any further information to resolve this issue. Thank you for your help!

**Comment by robert-hh** [comment]
It builds fine here with a Linux system. Did you clean the build before compiling? Either run "make clean" or remove the `build` directory.

**Comment by nawalter** [comment]
Thank you so much for your time helping with this!

I ran the command:
`rm -rf build-pca10056`

then from the ...ports\nrf folder, I successfully ran the command (at least there were no error messages):
`make submodules`

then ran the command:
`make BOARD=pca10056`
...and got a long log of build progress until the end where I got this:
`CC ../../py/bc.c
CC ../../py/showbc.c
CC ../../py/repl.c
CC ../../py/smallint.c
CC ../../py/frozenmod.c
C:/ProgramData/chocolatey/lib/make/tools/install/bin/make.exe -C C:/DESIGN/iron/Nordic_nrf52840/MicroPython/micropython/mpy-cross
make[1]: Entering directory 'C:/DESIGN/iron/Nordic_nrf52840/MicroPython/micropython/mpy-cross'
Use make V=1 or set BUILD_VERBOSE in your environment to increase build verbosity.
make[1]: Leaving directory 'C:/DESIGN/iron/Nordic_nrf52840/MicroPython/micropython/mpy-cross'
freeze error executing "modules/manifest.py": Error in manifest file: C:\DESIGN\iron\Nordic_nrf52840\MicroPython\micropython\ports\nrf\modules\manifest.py: Cannot stat C:\cygdrive\c\DESIGN\iron\Nordic_nrf52840\MicroPython\micropython\ports\nrf\modules\scripts\_mkfs.py
make: *** [../../py/mkrules.mk:189: build-pca10056/frozen_content.c] Error 1`

It looks like a file path was generated incorrectly:
`C:\cygdrive\c\DESIGN\iron\Nordic_nrf52840\MicroPython\micropython\ports\nrf\modules\scripts\_mkfs.py`

I verified the "_mkfs.py" file exists in the folder:
`C:\DESIGN\iron\Nordic_nrf52840\MicroPython\micropython\ports\nrf\modules\scripts`

**Comment by dpgeorge** [comment]
> It looks like a file path was generated incorrectly:
> `C:\cygdrive\c\DESIGN\iron\Nordic_nrf52840\MicroPython\micropython\ports\nrf\modules\scripts\_mkfs.py`

That path is generated from `$(PORT_DIR)` and `$(PORT_DIR)` comes from evaluating `$(shell pwd)`.

If you run `pwd` from the prompt when you are in the `ports/nrf` directory, what does it give?

**Comment by nawalter** [comment]
Here is what I get:
`C:\DESIGN\iron\Nordic_nrf52840\MicroPython\micropython\ports\nrf>pwd
/cygdrive/c/DESIGN/iron/Nordic_nrf52840/MicroPython/micropython/ports/nrf`

**Comment by jimmo** [comment]
@nawalter Can you try changing `$(shell pwd)` in `py/mkrules.mk` line 189 to `$(CURDIR)`.

Because the makefile doesn't change the working dir, these two should be equivalent.

**Comment by nawalter** [comment]
Thank you @dpgeorge and @jimmo for your help with this.

I was able to get  farther along, but ran into another issue running the `make BOARD=pca10056` command where it seems to be looking for the directory:
`build-pca10056/lib/nrfx/driver`
which doesn't exist.

However this directory does exist:
`build-pca10056/lib/nrfx/drivers`

The contents of the above `drivers` folder is a `src` folder containing 17 .p files and 17 .o files, as well as a folder `prs` containing one .p file and one .o file.

For kicks, I tried renaming the folder from `drivers` to `driver` to see if it would work, but I got a "Permission denied" error.
